src/bootstrap/dist.rs

```diff
@@ -98,6 +98,137 @@ pub fn docs(build: &Build, stage: u32, host: &str) {
     }
 }
 
+fn find_files(files: Vec<&str>, path: &Vec<PathBuf>) -> Vec<PathBuf> {
+    let mut found = Vec::new();
+
+    for file in &files {
+        let file_path =
+            path .iter()
+                .map(|dir| dir.join(file))
+                .find(|p| fs::metadata(&p).is_ok());
+
+        if let Some(file_path) = file_path {
+            found.push(file_path);
+        } else {
+            panic!("Could not find '{}' in {:?}", file, path);
+        }
+    }
+
+    found
+}
+
+fn make_win_dist(rust_root: &Path, plat_root: &Path, target_triple: &str) {
+    //Ask gcc where it keeps its stuff
+    let gcc_out =
+        String::from_utf8(
+            Command::new("gcc.exe")
```

Comment:
With access in Rust, it may actually be best to query [`build.cc(target)`](https://github.com/rust-lang/rust/blob/39bcd6f425426bfacd7de9fe0378df4bd9263d00/src/bootstrap/lib.rs#L876-L879) to find the path to gcc here.